relayer/canonical_validator_client.go

```diff
@@ -34,10 +34,51 @@ func (v *CanonicalValidatorClient) GetSubnetID(ctx context.Context, chainID ids.
 	return v.client.ValidatedBy(ctx, chainID)
 }
 
+// Gets the current validator set of the given subnet ID, include the validators' BLS public keys.
+// This implementation of GetValidatorSet currently makes two RPC requests, one to get the
+// subnet validators, and another to get their BLS public keys. This is necessary in order to enable
+// the use of the public APIs (which don't support "GetValidatorsAt") because BLS keys are currently
+// only associated with primary network validation periods. If ACP-13 is implementated in the future
+// (https://github.com/avalanche-foundation/ACPs/blob/main/ACPs/13-subnet-only-validators.md), it may
+// become possible to reduce this to a single RPC request that returns both the subnet validators
+// as well as their BLS public keys.
 func (v *CanonicalValidatorClient) GetValidatorSet(
 	ctx context.Context,
 	height uint64,
 	subnetID ids.ID,
 ) (map[ids.NodeID]*validators.GetValidatorOutput, error) {
-	return v.client.GetValidatorsAt(ctx, subnetID, height)
+	// Get the current subnet validators. These validators are not expected to include
```

Comment:
Rather than fully replacing the call to `GetValidatorsAt`, what are your thoughts on adding this workaround as either a configurable switch, or a fallback option in the case that `GetValidatorsAt` fails? 

For a switch, we could hardcode the public API endpoint and compare the config value against that. However, there's no guarantee that `GetValidatorsAt` is supported on some other node.

We could always try calling `GetValidatorsAt` then fallback to this workaround on failure, but that's strictly worse in terms of network round trips if we're using the public API.

Reply:
We could also add a config option indicating which API methods are supported by the configured node and switch on that. Alternatively, we could handle that flag dynamically by calling this method on startup and looking for the "method not supported" error message, though that would increase the set of things that could break unexpectedly.

I'm spitballing at this point, curious to hear your thoughts.

Reply:
(For context, the reason I'm in favor of having _some_ way to still call `GetValidatorsAt` is to have a path that minimizes latency for the relayer. If a project has a large enough number of validators, then they might want to run their own P-Chain API node to get that minimal latency path. That being said, I don't actually know if a call to `GetValidatorsAt` at the current height is necessarily faster than the workaround here, but I would suspect it is)

Reply:
All good thoughts...there are definitely trade offs one way or another at certain scales.

My main considerations are:
- The relative frequency of relayer that don't want to set up their own nodes to use an API, and their performance requirements.
- The relative frequency of relayer nodes that do want to set up their own nodes to use an API, and their performance requirements.
- Code and configuration complexity.

I think this is a minute enough of a detail that adding configuration options for it is not worth the additional code complexity and mental bandwidth of those setting up nodes to try to properly understand what the proper value for their set up would be. Checking whether or not the API method is available at start up is a cool idea to get around this, but there's no guarantee than an API won't change it behavior at any point in time, so would be prone to breaking.

That makes me think either always using the workaround method, or always trying to first use `GetValidatorsAt` and then fallback to the workaround if needed is likely the best approach. For relayer deployments without access to their own API nodes, my inclination is that an additional request would not be a critical performance decrease. I would also guess that deployments that do have or set up their own API nodes, tend to have strictly performance expectations/requirements. Given that, I think using this option as a fallback in the even that `GetValidatorsAt` fails is likely the best route forward.

Reply:
That sounds good to me. Very good point that a relayer operator concerned with latency would likely roll their own API node; if they choose not to, then one extra round trip is likely not going to make a difference to them.